stable/redis-ha/templates/redis-ha-configmap.yaml

```diff
@@ -13,134 +13,84 @@ data:
             sentinel_update "{{ template "redis-ha.fullname" . }}-server-0.{{ template "redis-ha.fullname" . }}"
             redis_update "{{ template "redis-ha.fullname" . }}-server-0.{{ template "redis-ha.fullname" . }}"
-{{- if .Values.redis.auth }}
-            auth "redis" "masterauth"
-            auth "sentinel"
-{{- end }}
         fi
     }
     function find_master(){
+        echo "Attempting to find master"
         if [[ ! `redis-cli -h $MASTER ping` ]]; then
-           echo "Cannot contact master, replacing dead master with default master"
-           setup_defaults
+           echo "Can't ping master, attempting to force failover"
+           if redis-cli -h {{ template "redis-ha.fullname" . }} -p {{ .Values.sentinel.port }} sentinel failover {{ .Values.redis.masterGroupName }} | grep -q 'NOGOODSLAVE' ; then 
+               setup_defaults
+               exit 0
+           fi
+           sleep 10
+           MASTER=`redis-cli -h {{ template "redis-ha.fullname" . }} -p {{ .Values.sentinel.port }} sentinel get-master-addr-by-name {{ .Values.redis.masterGroupName }} | grep -E '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}'`
+           if [[ "$MASTER" ]]; then
+               sentinel_update $MASTER
+               redis_update $MASTER
+           else
+              echo "Could not failover, exiting..."
+              exit 1
+           fi
         else
             echo "Found reachable master, updating config"
             sentinel_update $MASTER
             redis_update $MASTER
         fi
     }
-    function patch_conf(){
-        if [[ "$GROUP_NAME" != "{{ .Values.redis.masterGroupName }}" ]]; then
-            echo "Removing references to previous group name"
-            sed -i "s/ $GROUP_NAME / {{ .Values.redis.masterGroupName }} /" $SENTINEL_CONF
-        fi
-        {{- range $key, $value := .Values.redis.config }}
-        sed -i "s/^.*{{ $key }}.*/{{ $key }} {{ $value }}/" $REDIS_CONF
-        {{- end }}
-        {{- $root := . -}}
-        {{- range $key, $value := .Values.sentinel.config }}
-        sed -i "s/^.*sentinel {{ $key }}.*/sentinel {{ $key }} {{ $root.Values.redis.masterGroupName }} {{ $value }}/" $SENTINEL_CONF
-        {{- end }}
-    }
     mkdir -p /data/conf/
     echo "Initializing config.."
 
-    if [[ ! -f $REDIS_CONF && ! -f $SENTINEL_CONF ]]; then
-        cp /readonly-config/redis.conf $REDIS_CONF
-        cp /readonly-config/sentinel.conf $SENTINEL_CONF
-        if [[ "$MASTER" ]]; then
-            find_master
-        else
-            setup_defaults
-        fi
-    elif [[ "$MASTER" == "" ]]; then
-        echo "Found configs but no redis master found! Configuring default master.."
-        patch_conf
-        setup_defaults
-    else
-        patch_conf
+    cp /readonly-config/redis.conf $REDIS_CONF
+    cp /readonly-config/sentinel.conf $SENTINEL_CONF
+
+    if [[ "$MASTER" ]]; then
         find_master
+    else
+        setup_defaults
+    fi
+    if [[ "$AUTH" ]]; then
```

Comment:
@ssalaues you probably want to add this routine to `setup_defaults`, otherwise it won't get invoked after https://github.com/helm/charts/pull/7323/commits/4984e0d8da5c56e9ffe97aba54e92ce02936fe47#diff-a988587cdea159026a58cc36592ce812R64